Week 5

orionkoepke opened the pull request.

orionkoepke committed:
Bonetrousle Added

orionkoepke committed:
Made Small Change to Bonetrousle

I changed the top of the README file. I deleted the tags and put
"difficulty : " infront of the difficulty.

isaac-gs commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -0,0 +1,97 @@
+# Bonetrousle
+[https://www.hackerrank.com/challenges/bonetrousle](https://www.hackerrank.com/challenges/bonetrousle)
+
```
What type or category does this fall into?

isaac-gs commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -0,0 +1,97 @@
+
+## Problem
+### Statement
+Once upon a time, Papyrus the skeleton went to buy some pasta from the store. The store's inventory is bare-bones and they only sell one thing — boxes of uncooked spaghetti! The store always stocks exactly k boxes of pasta, and each box is numbered sequentially from 1 to k. This box number also corresponds to the number of sticks of spaghetti in the box, meaning the first box contains 1 stick, th ...
```
This is directly copied from Hackerrank, according to their user agreement we are not allowed to do that. Please summarize the essence of the problem, the key ideas involved, rather than copying the problem statement.

isaac-gs commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -0,0 +1,97 @@
+
+###Input Format
+
+The first line contains a single integer, t, denoting the number of trips to the store.
```
Also copied. Please summarize this in a short and clear manner.

isaac-gs commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -0,0 +1,97 @@
+
+###Constraints
+
+* 1 <= t <= 20
```
Don't worry about this, we are assuming the user is going to actually read the problem. Get rid of the "###Constraints" section.

isaac-gs commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -0,0 +1,97 @@
+
+* b <= k
+
+###Output Format
```
Get rid of this section. Assume user reads the problem.

isaac-gs commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -0,0 +1,97 @@
+
+If there are multiple possible solutions, you can print any one of them. Do not print any leading or trailing spaces.
+
+### Overview
```
Much better way of summarizing the problem. I like this.

isaac-gs commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -0,0 +1,97 @@
+
+## Conclusion
+<p>
+It is important to keep in mind that for this problem some variable values could exceed
```
Are there any takeaways you have from working on this problem? Maybe for similar problems?

isaac-gs commented on Guide/Constructive/Bonetrousle/solution.java:
```diff
@@ -0,0 +1,64 @@
+import java.io.*;
+import java.util.Scanner;
+
+/**
```
Unnecessary to include, that's why we use git. It tracks contributions.

isaac-gs commented on Guide/Constructive/Bonetrousle/solution.java:
```diff
@@ -0,0 +1,64 @@
+ */
+public class Bonetrousle
+{
+    public static void main(String[] args) throws IOException
```
Could you break this code up to make it cleaner? Split it into functions a little?

isaac-gs reviewed (changes requested):
Looks pretty good but please stick to the template that we designed. See comments.

orionkoepke committed:
Corrected README.md and solution.java in Bonetrousle

I fixed all the problems Isaac mentioned in his comment.

orionkoepke commented:
@ASAAR I fixed it, but I am not sure how to do the pull request again.

NilsG-S commented:
@orionkoepke You don't have to. When you committed your code to the branch the pull request was automatically updated.

asclines commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -62,6 +32,12 @@ formula above so that it does work. Note that the problem says that any combinat
 the needed number of noodles is a valid solution.
 </p>
```
Whats with the HTML?

orionkoepke replied:
It allowed me to put comment on multiple lines without them appearing as different paragraphs.

asclines commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -91,7 +67,9 @@ the list of boxes out. This loop goes b times and thusly the algorithm's time co
-It is important to keep in mind that for this problem some variable values could exceed
-the value of an int in Java. Also, System.out.println(), may be too slow for big inputs.
-Instead, use OutputStream.
+As computer science is closely related to math, some problems can be easily solved 
```
I like this conclusion

asclines commented on Guide/Constructive/Bonetrousle/README.md:
```diff
@@ -62,6 +32,12 @@ formula above so that it does work. Note that the problem says that any combinat
 
+<p>
+It is important to keep in mind that for this problem some variable values could exceed
+the value of an int in Java. Also, System.out.println(), may be too slow for big inputs.
```
When using code syntax in markdown, format it like so: `System.out.println()` to make things easier

orionkoepke replied:
How did you get the code to look like that? Online it says to use `code`, but that wasn't doing anything on my computer.

orionkoepke replied:
Never mind. It let me do it once I took it out of the paragraph tags.

asclines reviewed (commented):
The code looks good!

orionkoepke committed:
Changed formating

I changed System.out.println() to look like code. It was one of Alex's
suggestions.
